src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        const index = Animal.alive.indexOf(animal);
+
+        Animal.alive.splice(index, 1);
```

Comment:
Yep, and one  more short way:

```suggestion
        Animal.alive = Animal.alive.filter(aliveAnimal => aliveAnimal !== animal)
```